packages/resolve-query/src/read-model.js

```diff
@@ -107,7 +111,13 @@ const createReadModel = ({ projection, eventStore, adapter }) => {
   const builtProjection = projection
     ? currentAdapter.buildProjection(projection)
     : null
-  const repository = {}
+  const externalEventListeners = []
+  const repository = {
+    eventListener: event =>
+      externalEventListeners.forEach(callback =>
+        Promise.resolve().then(callback.bind(null, event))
```

Comment:
Why?
Promise.resolve().then(callback.bind(null, event)) === callback.bind(null, event), or not?

Reply:
It's not same for two reasons. Lying on surface, is first expression is promise invocation, and second is declare bound function exotic object and do nothing with it - but reason, of course, id not about it.
Such expression forces callback function to be invoked from dedicated tick domain callback, which helps to segregate event handling in projections and DIFFs publishing process.
DIFF publish process can fail to some reasons, for example, problems with socket, and it should NOT affect projection side